src/Illuminate/Http/Resources/ConditionallyLoadsAttributes.php

```diff
@@ -66,22 +67,29 @@ protected function mergeData($data, $index, $merge, $numericKeys)
      * Remove the missing values from the filtered data.
      *
      * @param  array  $data
-     * @param  bool  $numericKeys
      * @return array
      */
-    protected function removeMissingValues($data, $numericKeys = false)
+    protected function removeMissingValues($data)
     {
+        $numericKeys = true;
+
         foreach ($data as $key => $value) {
             if (($value instanceof PotentiallyMissing && $value->isMissing()) ||
                 ($value instanceof self &&
                 $value->resource instanceof PotentiallyMissing &&
                 $value->isMissing())) {
                 unset($data[$key]);
             }
+            $numericKeys = $numericKeys && is_numeric($key);
+        }
+
+        if (property_exists($this, 'preserveKeys') && $this->preserveKeys === true) {
+            $values = array_values($data);
```

Comment:
The result of `$values === $data` is not relevant here, we return the same data either way. 

We can replace both lines with `return $data;`.

Reply:
You are correct, fixed it.